etc/sync-function-template.js

```diff
@@ -796,14 +796,34 @@ function synctos(doc, oldDoc) {
 
   var theDocDefinition = docDefinitions[theDocType];
 
+  if (theDocDefinition.customActions && typeof(theDocDefinition.customActions.onTypeIdentificationSucceeded) === 'function') {
+    theDocDefinition.customActions.onTypeIdentificationSucceeded(doc, oldDoc);
+  }
+
   authorize(doc, oldDoc, theDocDefinition);
 
+  if (theDocDefinition.customActions && typeof(theDocDefinition.customActions.onAuthorizationSucceeded) === 'function') {
+    theDocDefinition.customActions.onAuthorizationSucceeded(doc, oldDoc);
+  }
+
```

Comment:
That's an interesting suggestion. I was trying to keep it simple though. And I didn't want to get too fancy because I'm planning a follow-up PR that adds a third parameter to the custom action events: an object that is passed from one to the next and that has metadata from each step of the process added to it along the way. Admittedly, that wouldn't preclude your suggestion, but I'll keep it in mind for that future PR instead.